src/stirling/source_connectors/socket_tracer/protocols/common/data_stream_buffer.h

```diff
@@ -95,7 +95,21 @@ class DataStreamBuffer {
    * @param pos The logical position of the data.
    * @return The timestamp or error if the position does not contain valid data.
    */
-  StatusOr<uint64_t> GetTimestamp(size_t pos) const { return impl_->GetTimestamp(pos); }
+  StatusOr<uint64_t> GetTimestamp(size_t pos) {
```

Comment:
Adding a chunk with a non monotonic timestamp through `DataStream::Add` should be easy to test this behavior. Can we add an new test case to [DataStreamBufferTest.Timestamp](https://github.com/pixie-io/pixie/blob/3e02189116d9c3c5f488504894d015809d0bcf19/src/stirling/source_connectors/socket_tracer/protocols/common/data_stream_buffer_test.cc#L131)?

Reply:
Added!